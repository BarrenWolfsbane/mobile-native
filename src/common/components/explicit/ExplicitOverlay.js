import React, {
  Component
} from 'react';

import {
  Text,
  TouchableOpacity,
  StyleSheet,
  View,
} from 'react-native';

import { observer } from 'mobx-react/native';
import { CommonStyle as CS } from '../../../styles/Common';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { GOOGLE_PLAY_STORE } from '../../../config/Config';

/**
 * Explicit overlay
 */
@observer
export default class ExplicitOverlay extends Component {

  /**
   * Default props
   */
  static defaultProps = {
    iconSize: 80,
    hideText: false,
    iconPosition: 'right'
  };

  /**
   * toggle overlay
   */
  toogle = () => {
    this.props.entity.toggleMatureVisibility();
  }

  /**
   * Render
   */
  render() {
    const {
      iconSize,
      hideText,
      fontStyle,
      containerStyle,
      closeContainerStyle,
      iconPosition
    } = this.props;

    if (this.props.entity.mature_visibility) {
      const size = Math.max(iconSize/2.5, 25);
      const styles = iconPosition == 'right' ?
        [ CS.positionAbsoluteTopRight, CS.paddingRight4x, CS.marginRight4x] :
        [ CS.positionAbsoluteTop, CS.paddingLeft4x, CS.marginLeft4x]
      return (
        <View style={[styles, closeContainerStyle]}>
          <Icon name="explicit" size={size} color={'red'} style={CS.shadow} onPress={this.toogle}/>
        </View>
      )
    }

    const text = GOOGLE_PLAY_STORE ? 'This post cannot be shown' : 'Confirm you are 18+';

    return (
      <TouchableOpacity activeOpacity={1} style={[CS.positionAbsolute, CS.centered, CS.backgroundDarkGreyed, styles.onTop, containerStyle]} onPress={this.toogle}>
        <Icon name="explicit" size={iconSize} color={'white'} style={CS.shadow}/>
        {!hideText && <Text style={[CS.colorWhite, CS.shadow, fontStyle]}>{text}</Text>}
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  onTop: {
    zIndex: 99999
  }
})