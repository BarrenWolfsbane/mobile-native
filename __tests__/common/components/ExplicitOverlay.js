import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';
import ExplicitOverlay from '../../../src/common/components/explicit/ExplicitOverlay';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';

describe('Explicit overlay component', () => {

  let user, comments, entity, screen;
  beforeEach(() => {

    let mockResponse = activitiesServiceFaker().load(1);
    entity = mockResponse.activities[0];
    
    entity.mature_visibility = true;
    screen = shallow(
      <ExplicitOverlay entity={entity}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a TouchableOpacity', async () => {
    screen.update();

    expect(screen.find('TouchableOpacity')).toHaveLength(1);
  });


  it('change visibility', async () => {
    let mockResponse = activitiesServiceFaker().load(1);
    entity = mockResponse.activities[0];
    entity.mature_visibility = true;
    entity.toggleMatureVisibility = jest.fn();
    screen = shallow(
      <ExplicitOverlay entity={entity}/>
    );

    screen.find('TouchableOpacity').simulate('press');
    screen.update();

    expect(entity.toggleMatureVisibility).toHaveBeenCalled();
  });
});
