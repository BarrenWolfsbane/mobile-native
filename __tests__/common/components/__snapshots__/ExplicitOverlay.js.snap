// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explicit overlay component renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <Icon
    allowFontScaling={false}
    color="red"
    name="explicit"
    size={32}
    style={
      Object {
        "elevation": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 0.5,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 0.5,
      }
    }
  />
</TouchableOpacity>
`;
